### ISSUE TITLE
Why are empty arrays encoded as nil?

### ISSUE THREAD

**Issue by nuudles** [hkellaway/Gloss#68]
Out of curiosity, why are empty arrays encoded as nil as seen on this line:

```
return encodedArray.isEmpty ? nil : [key : encodedArray]
```

Wouldn't you want the encoded JSON to closely mirror what's being encoded?

**Comment by hkellaway** [comment]
I guess the original thought behind that was that is the array was empty, the JSON encodings had failed - and returning `nil` in that case would be more consistent. But, thinking about it now, that is probably too opinionated for that call to be

**Comment by nuudles** [comment]
Yeah, I think it makes more sense to encode it as an empty JSON array. In my case, I have a non-optional array property of my struct, so it currently fails since the encoded JSON doesn't include the empty array. I end up having to do something like:

```
self.array = "array" <~~ json ?? []
```

Which is less than ideal.

**Comment by hkellaway** [comment]
Thanks for pointing that out - it'll be fixed next version

**Comment by hkellaway** [comment]
fixed as of version `0.6.0` :tada:

**Comment by nuudles** [comment]
Awesome, thanks!
